stl/inc/algorithm

```diff
@@ -10163,16 +10163,24 @@ _NODISCARD constexpr const _Ty& clamp(const _Ty& _Val, const _Ty& _Min_val, cons
         return _Val;
     }
 #endif // _ITERATOR_DEBUG_LEVEL == 2
+    if constexpr ((is_integral_v<_Ty> || is_enum_v<_Ty> || is_pointer_v<_Ty>
+#ifdef _M_FP_FAST
+                      || is_floating_point_v<_Ty>
+#endif //  _M_FP_FAST
+                      ) &&(is_same_v<_Remove_cvref_t<_Pr>, less<>> || is_same_v<_Remove_cvref_t<_Pr>, less<_Ty>>) ) {
+        // Avoid short-circuit evaluation to promote cmov generation; see GH-2334
+        return (_STD min) ((_STD max) (_Val, _Min_val), _Max_val);
+    } else {
+        if (_DEBUG_LT_PRED(_Pred, _Max_val, _Val)) {
+            return _Max_val;
+        }
 

```

Comment:
Style nitpick: I would prefer for this empty line to be kept, as we're beginning a new chain of control flow.